neurom/core/_neuron.py

```diff
@@ -152,6 +154,43 @@ def graft_neuron(root_section):
     return Neuron(soma=Soma(root_section.points[:1]), neurites=[Neurite(root_section)])
 
 
+def sample_morph_points(morph, sample_distance):
+    """Sample points along the morphology.
+
+    Args:
+        morph (neurom.FstNeuron): morphology
+        sample_distance (int in um): points sampling distance
+
+    Returns:
+        Dict: sampled points per neurite. Points are of shape (N, 3) where N is the number of
+        sampled points.
+    """
+    # map of section to its remaining offset
+    section_offsets = {}
+    morph_points = defaultdict(list)
+    for section in iter_sections(morph):
+        if section.parent is None:
+            parent_section_offset = 0
+        else:
+            parent_section_offset = section_offsets[section.parent.id]
+        segment_offset = parent_section_offset
+        for segment in iter_segments(section):
+            segment_len = morphmath.segment_length(segment)
+            if segment_offset + segment_len < sample_distance:
+                segment_offset += segment_len
+            elif segment_offset + segment_len == sample_distance:
+                morph_points[section.type].append(segment[1][COLS.XYZ])
+                segment_offset = 0
+            else:
+                offsets = np.arange(sample_distance - segment_offset, segment_len, sample_distance)
+                for offset in offsets:
+                    morph_points[section.type].append(
+                        morphmath.linear_interpolate(*segment, offset / segment_len))
+                segment_offset = segment_len - offsets[-1]
+        section_offsets[section.id] = segment_offset
+    return {neurite: np.vstack(points) for neurite, points in morph_points.items()}
```

Comment:
Not sure why the output is a dictionary, could the neurite selection correspond to a neurite type filtering instead?

Reply:
Or return a new morphology object with the subsampled neurites so that it can be used for any feature extraction downstream.

Reply:
How about deepcopying the morphology, iterating over the sections as you do and for each section create a mask with the rows that will remain in the section (taking care not to remove the start and end) and then mutating the point data?

Reply:
@lidakanari yeah that does seem better

Reply:
@eleftherioszisis that seems to me like something that should be done on morph-tool's end. For the in-situ validations we are interested in the positions themselves.